Add snap package creation support

popey opened the pull request:
Hi! I recently discovered goxel, and while I lack the artistic talent to create really great voxel based images, it's a great app!

I noticed that you don't have native linux packages, so I thought I'd add support for snap packages. Snaps are universal packages for multiple Linux distributions. With this pull request, you can build a snap of goxel and provide it automatically to millions of users via the snap store. A single snap works on Ubuntu LTS releases (14.04, 16.04) and interim releases (17.04, upcoming 17.10), Debian, Fedora, and more.

We have a command line tool called snapcraft which can be used to build the snap (which is how I built a snap and tested on my Ubuntu 16.04 system). We also have a free [build servic](http://build.snapcraft.io/)e you can hook up to this github repo, and build automatically on each commit, uploading to the store with each change.

If you accept this PR, then simply create an account in the snap store at http://dashboard.snapcraft.io, register your new snap 'goxel' and then use http://build.snapcraft.io to hook up github to the store, and you're done.

Builds will land in the edge channel on every commit, for adventurous users. When you create a stable release, you can push that to the stable channel for more conservative users.

This is all documented at http://snapcraft.io/docs but I'm happy to answer any questions here, on irc (popey), telegram (popeydc) or email (alan.pope@canonical.com).

Thanks!

guillaumechereau commented:
Thanks that looks interesting.  Two questions:

- Why did you remove the desktop file?
- goxel can link to libpng if available, or else it uses stb_image as a fallback.  I would like to make sure that libpng is used in the snap, because stb_image is much slower, how to do that?

popey commented:
"Why did you remove the desktop file?"

Because I suck at git :) Sorry. It needs that.

"goxel can link to libpng if available, or else it uses stb_image as a fallback. I would like to make sure that libpng is used in the snap, because stb_image is much slower, how to do that?"

Add "libpng12-dev" to the stage-packages stanza. That should do it. I'll update this PR.

popey commented:
Ok, checked and there is a desktop file in there, I had removed the one that was in the wrong folder, there should still be one in snap/gui. I added libpng too, and it builds correctly:-

```
Building goxel 
scons debug=0
scons: Reading SConscript files ...
Checking whether __GLIBC__ is declared... yes
Checking for C library libpng... yes
scons: done reading SConscript files.
scons: Building targets ...
```

guillaumechereau commented:
OK I merged all your changes as a single commit.
Thanks!

popey commented:
Sweet! Now if you go to build.snapcraft.io it can create amd64, i386 and armhf builds and put them directly in the store! We can put out a call for testing once you do.

guillaumechereau commented:
I opened an account in snapcraft.io but the process failed when I tried to register the name 'goxel', with the error message: "forbidden".  I am not sure what is the problem.

popey commented:
Sorry for the delay replying, I missed the notification. What happens if you go to https://dashboard.snapcraft.io/dev/snaps/register-name/ and attempt to register the goxel name?

popey commented:
On the subject of the 'forbidden' message. If you have some time to help us debug that, it would help tremendously. If you could repeat what you did to register the name via build.snapcraft.io, but in the browser have the javascript console open at the networking tab. When the error occurs, refresh and capture the data. If you could file an issue at https://github.com/canonical-websites/build.snapcraft.io/issues or perhaps add to this existing issue which looks related https://github.com/canonical-websites/build.snapcraft.io/issues/792 - we'd appreciate that.

guillaumechereau commented:
OK I checked with the debugger on, and the register-name call returns more information in the json:
"message": "Developer profile is missing namespace."
"code": "user-not-ready"

So I guess I have to update my profile in ubuntu one first...

guillaumechereau commented:
The documentation doesn't really address what the developer namespace should be.  I guess it doesn't matter and I can pick any random name, but is there some kind of standard for what this value should be?

popey commented:
Ugh, really sorry about that. Yes, go to https://dashboard.snapcraft.io/dev/account/ and fill in a namespace. Personally I use my github Id "popey", but you can use whatever.

guillaumechereau commented:
OK, seems to work now! Thanks.
